Experimental: Introduce a pool of query planners (#4897)
apollographql/router: .changesets/exp_carton_ginger_magnet_beacon.md

### Experimental: Introduce a pool of query planners ([PR #4897](https://github.com/apollographql/router/pull/4897))

The router supports a new experimental feature: a pool of query planners to parallelize query planning.

You can configure query planner pools with the `supergraph.query_planner.experimental_available_parallelism` option:

```yaml
supergraph:
  query_planner:

>     experimental_available_parallelism: auto # number of available cpus

Author: Fixed in [d492701](https://github.com/apollographql/router/pull/4897/commits/d492701bbb618abf50d76a9e1f2909854e781b2d)

```

> Its value is the number of query planners that run in parallel, and its default value is `1`. You can set it to the special value `auto` to automatically set it equal to the number of available CPUs.

Author: Good question!

My mac has 8 cpus available, and if i go back and forth from 8 to auto, it will reload at the moment.

### Experimental: Introduce a pool of query planners ([PR #4897](https://github.com/apollographql/router/pull/4897))

> This changeset introduces an experimental pool of query planners to parallelize query planning.

Reviewer: ```suggestion
The router supports a new experimental feature: a pool of query planners to parallelize query planning.
```

> This feature is experimental, you can discuss it by following this link: https://github.com/apollographql/router/discussions/4917
> 
> 
> Configuration:

Reviewer: ```suggestion
You can configure query planner pools with the `supergraph.query_planner.experimental_available_parallelism` option:
```

> This feature is experimental, you can discuss it by following this link: https://github.com/apollographql/router/discussions/4917
> 
> 
> Configuration:
> 
> ```yaml
> supergraph:
>   query_planner:
>     experimental_available_parallelism: auto # number of available cpus
> ```
> 
> Note you can also set `experimental_available_parallelism` to a number representing how many planners you want to use in a pool.
> The default is `1`.

Reviewer: ```suggestion
This feature is experimental, you can discuss it by following this link: https://github.com/apollographql/router/discussions/4917

You can configure query planner pools by setting `supergraph.query_planner.experimental_available_parallelism`:

```yaml
supergraph:
  query_planner:
    experimental_available_parallelism: auto # number of available cpus
```

Its value is the maximum number of query planners that can run in parallel, and its default value is `1`. You can set it to the special value `auto` to automatically set it equal to the number of available CPUs. 
```

> Note you can also set `experimental_available_parallelism` to a number representing how many planners you want to use in a pool.
> The default is `1`.
> 

Reviewer: ```suggestion
Its value is the maximum number of query planners that can run in parallel, and its default value is `1`. You can set it to the special value `auto` to automatically set it equal to the number of available CPUs.

You can discuss and comment about query planner pools in this [GitHub discussion](https://github.com/apollographql/router/discussions/4917).

```
